src/protocolgame.cpp

```diff
@@ -916,7 +1001,11 @@ void ProtocolGame::parseSay(NetworkMessage& msg)
 		return;
 	}
 
-	addGameTask(&Game::playerSay, player->getID(), channelId, type, receiver, text);
+	if (channelId == CHANNEL_CAST) {
+		g_dispatcher.addTask(createTask(std::bind(&ProtocolGame::sendChannelMessage, this, player->getName(), text, TALKTYPE_CHANNEL_R1, channelId)));
```

Comment:
Your code in connections.cpp and outputmessage.cpp caused closeConnection() and closeSocket() on my server with 100 players online. I thought unnecessary to create a new protocol -> protocolcast.cpp.